jlm/llvm/ir/operators/Load.hpp

```diff
@@ -24,12 +24,52 @@ namespace jlm::llvm
  */
 class LoadOperation : public rvsdg::SimpleOperation
 {
+public:
+  class MemoryStateInputIterator final : public rvsdg::input::iterator<rvsdg::SimpleInput>
+  {
+  public:
+    constexpr explicit MemoryStateInputIterator(rvsdg::SimpleInput * input)
+        : rvsdg::input::iterator<rvsdg::SimpleInput>(input)
+    {}
+
+    [[nodiscard]] rvsdg::SimpleInput *
+    next() const override
+    {
+      auto index = value()->index();
+      auto node = value()->node();
+
+      return node->ninputs() > index + 1 ? node->input(index + 1) : nullptr;
```

Comment:
This > operator feels a bit yoda-like, I personally think it reads more naturally the other way around

Reply:
Fixed. Does this work for you?